### PR TITLE
add parameter r

### LINKED ISSUE #11: Parameter -r

### PR DESCRIPTION
**Pull request by justinmueller05082000**
fixes #11

### CODE REVIEW

**Review by Al2Klimov — CHANGES_REQUESTED** [review]
```
/trainee-go-grep # mkdir a
/trainee-go-grep # echo c > a/b
/trainee-go-grep # go run . -r -e c a
open ab: no such file or directory
exit status 2
/trainee-go-grep #
```

File: main.go [hunk 1]
```diff
@@ -163,6 +164,10 @@ func compareAndPrint(file *os.File) {
 }
 
 func recursiveSearch(dirName string) {
+	if !strings.HasSuffix(dirName, "/") {
+		dirName = dirName + "/"
```
Comment (Al2Klimov) [inline main.go]: Any way to shorten this line?

**Review by Al2Klimov — CHANGES_REQUESTED** [review]

**Review by Al2Klimov — CHANGES_REQUESTED** [review]
What does grep do if you specify -r, but no files?

**Comment by justinmueller05082000** [comment]
i guess it goes trough the directory u are in at the moment

**Comment by Al2Klimov** [comment]
And your grep?

**Comment by justinmueller05082000** [comment]
exit code 1

**Comment by Al2Klimov** [comment]
Not good.

File: main.go [hunk 2]
```diff
@@ -81,8 +81,17 @@ func main() {
...
 	if *recursiveParameter {
-		for _, dirName := range flag.Args() {
-			recursiveSearch(dirName)
+		if len(flag.Args()) < 1 {
+			currentPath, currentPathErr := os.Getwd()
+			if currentPathErr != nil {
+				fmt.Fprintln(os.Stderr, currentPathErr)
+				os.Exit(2)
+			}
+			recursiveSearch(currentPath)
```
Comment (Al2Klimov) [inline main.go]: Any way not to need `currentPath` here?

**Review by Al2Klimov — CHANGES_REQUESTED** [review]

File: main.go (hunk 2 above)
Comment (justinmueller05082000) [reply main.go]: append current path to flag.Args() if len(flag.Args()) < 1?

Comment (Al2Klimov) [reply main.go]: You're in `/foo/bar`. You'd like to pass `/foo/bar` to a program. How to shorten `/foo/bar`?

Comment (justinmueller05082000) [reply main.go]: */bar?

Comment (Al2Klimov) [reply main.go]: No, because there isn't any `/foo/bar/*/bar`.

Comment (justinmueller05082000) [reply main.go]: i dont know how to shorten it then

Comment (justinmueller05082000) [reply main.go]: pwd shows me my current path

Comment (justinmueller05082000) [reply main.go]: but i used os.Getwd() to get my current path i dont understand how to shorten it even more

Comment (Al2Klimov) [reply main.go]: You're effectively going to read **the current directory**. How to do that as short as possible?

Comment (justinmueller05082000) [reply main.go]: as I do at the moment

Comment (Al2Klimov) [reply main.go]: .

Comment (justinmueller05082000) [reply main.go]: i dont know how niko also does not know it

Comment (Al2Klimov) [reply main.go]: Tip: .

**Review by Al2Klimov — CHANGES_REQUESTED** [review]
* [x] `git reset --soft HEAD~$(git rev-list --count $(git merge-base HEAD master)..HEAD^)`
* [x] `git commit --amend --no-edit`
* [x] `git push -f origin feature/add-parameter-r-11`

**Review by Al2Klimov — CHANGES_REQUESTED** [review]
Write a test protocol.

**Comment by justinmueller05082000** [comment]
**Test protocol:**

**Path where the files are in:** `~/go/src/github.com/justinmueller05082000/trainee-go-grep/txts/2nddir/`

**2 Files:**

**1st. toGrep.txt:**
```
zebra
zebra
dog
zebra
cat
```
**2nd. toGrep2.txt:**
```
dog
cat
cat
mouse
zebra
```

**Case 1:**

**Terminal:** `go run . -r -e 'zebra' ~/go/src/github.com/justinmueller05082000/trainee-go-grep/`
**Output:**
```
/home/jmueller/go/src/github.com/justinmueller05082000/trainee-go-grep/txts/2nddir/toGrep.txt:zebra
/home/jmueller/go/src/github.com/justinmueller05082000/trainee-go-grep/txts/2nddir/toGrep.txt:zebra
/home/jmueller/go/src/github.com/justinmueller05082000/trainee-go-grep/txts/2nddir/toGrep.txt:zebra
/home/jmueller/go/src/github.com/justinmueller05082000/trainee-go-grep/txts/2nddir/toGrep2.txt:zebra
```

**Terminal:**` grep -r -e 'zebra' ~/go/src/github.com/justinmueller05082000/trainee-go-grep/`
**Output:**
```
/home/jmueller/go/src/github.com/justinmueller05082000/trainee-go-grep/txts/2nddir/toGrep.txt:zebra
/home/jmueller/go/src/github.com/justinmueller05082000/trainee-go-grep/txts/2nddir/toGrep.txt:zebra
/home/jmueller/go/src/github.com/justinmueller05082000/trainee-go-grep/txts/2nddir/toGrep.txt:zebra
/home/jmueller/go/src/github.com/justinmueller05082000/trainee-go-grep/txts/2nddir/toGrep2.txt:zebra
```

**Conclusion:**
All outputs are equal except for the colored highlighting in the terminal.

**Review by Al2Klimov — APPROVED** [review]

**Pull request merged**